libtransact/src/workload/runner.rs

```diff
@@ -554,7 +566,15 @@ pub fn submit_batches_from_source(
 
         // get next target, round robin
         next_target = (next_target + 1) % targets.len();
-        thread::sleep(time_to_wait);
+        let diff = time::Instant::now() - submission_start;
+        let submission_time = match submission_avg {
+            Some(val) => (diff + val) / 2,
+            None => diff,
+        };
+        submission_avg = Some(submission_time);
+
+        thread::sleep(time_to_wait - submission_time);
```

Comment:
This should also be updated to use time_to_wait.saturating_sub

Reply:
Yeah, pushed it and then saw the issue. Just pushed yet another fix. Thanks